---
description: >-
  If you run the workflow with default settings, these following files are
  expected
---

# Outputs Description





#### UMI Processing:

| File ID | Example | Description |
| :--- | :--- | :--- |
| **composite\_umi\_frequencies** | composite-umi-frequencies.txt | This is text file consisting of frequencies of unique molecular identifier as seen by Marianas ProcessLoopUMIFastq |
| **clipping\_info** | info.txt | File having information about all the clipped unique molecular identifiers from the fastq.gz files by Marianas ProcessLoopUMIFastq |

#### Standard Bam Processing:

| File ID | Example | Description |
| :--- | :--- | :--- |
| **standard\_bam\_indel\_realign\_targets** | seracare\_standard\_md.bed | Covered intervals targeted in Indel Realignment step |
| **standard\_bam\_alignment\_metrics** | seracare\_standard\_md.metrics | Picard alignment metrics for standard bam |
| **md\_bam** | seracare\_standard\_md.bam | Binary Alignment Map \(BAM\) File generated after marking duplicate reads using Picard MarkDuplicate tool |
| **clstats1** | seracare\_R1\_001\_umi-clipped.fastq.gz\_trimming\_report.txt | Trimming statistics generated by TrimGalore/Cutadapt for READ1 |
| **clstats2** | seracare\_R2\_001\_umi-clipped.fastq.gz\_trimming\_report.txt | Trimming statistics generated by TrimGalore/Cutadapt for READ2 |
| **bqsr\_bam** | seracare\_standard\_md\_abra\_fm\_bqsr.bam | Base Recalibrated Binary Alignment Map format file generated using GATK BaseRecalibrator and ApplyBQSR tool |

secondaryFiles:

| File ID | Example |
| :--- | :--- |
| **md\_bam index** | seracare\_standard\_md.bai |
| **bqsr\_bam index** | seracare\_standard\_md\_abra\_fm\_bqsr.bai |

#### 

#### Bam Collapsing:

| File ID | Example | Description |
| :--- | :--- | :--- |
| **unfiltered-bam** | seracare\_unfiltered\_abra\_fm.bam | This file is generated after collapsing of reads from the standard bam file. This is all duplex, simplex and singletons as part of the alignment |
| **simplex-bam** | seracare\_unfiltered-simplex.bam | This Simplex BAM file is generated from Marianas SeparateBams module which seprate bam file based on duplex and simple clusters |
| **duplex-bam** | seracare\_unfiltered-duplex.bam | This Duplex BAM file is generated from Marianas SeparateBams module which seprate bam file based on duplex and simple clusters |
| **second\_pass\_insertions** | second-pass-insertions.txt | This file containing insertion is generated by Marianas DuplexUMIBamToCollapsedFastqSecondPass |
| **second\_pass\_alt\_alleles** | second-pass-alt-alleles.txt | This file containing ALT ALLELES is generated by Marianas DuplexUMIBamToCollapsedFastqSecondPass |
| **intervals** | seracare\_standard\_md\_abra\_fm\_bqsr-intervals.txt |  |
| **pileup** |  |  |
| **pileup\_without\_duplicates** | seracare\_standard\_md\_abra\_fm\_bqsr-pileup-without-duplicates.txt |  |
| **intervals\_without\_duplicates** | seracare\_standard\_md\_abra\_fm\_bqsr-intervals-without-duplicates.txt |  |
| **unfiltered\_bam\_indel\_realigned\_targets** | seracare\_unfiltered.bed |  |
| **gzip\_read1** | seracare\_collapsed\_R1\_fastq.gz | This is the collapsed READ1 gzip fastq file generated after Marianas collapsing |
| **gzip\_read2** | seracare\_collapsed\_R2\_fastq.gz | This is the collapsed READ2 gzip fastq file generated after Marianas collapsing |
| **first\_pass\_insertions** | first-pass-insertions.txt | This file containing insertion is generated by Marianas DuplexUMIBamToCollapsedFastqFirstPass |
| **collapsed\_fastq\_1** | collapsed\_R1_\__.fastq | This is the collapsed Read 1 fastq file generated after Marianas collapsing |
| **collapsed\_fastq\_2** | collapsed\_R2\_.fastq | This is the collapsed Read 2 fastq file generated after Marianas collapsing |
| **alt\_allele\_file** | first-pass-alt-alleles.txt | This file containing ALT ALLELES is generated by Marianas DuplexUMIBamToCollapsedFastqFirstPass |
| **alignment\_metrics\_unfiltered** | seracare\_unfiltered\_alignment\_metrics.txt | Alignment metrics TXT file generated by Picard CollectAlignmentMetrics for Unfilered BAM File. |
| **alignment\_metrics\_simplex** | seracare\_unfiltered-simplex\_alignment\_metrics.txt | Alignment metrics TXT file generated by Picard CollectAlignmentMetrics for SIMPLEX BAM File |
| **alignment\_metrics\_duplex** | seracare\_unfiltered-duplex\_alignment\_metrics.txt | Alignment metrics TXT file generated by Picard CollectAlignmentMetrics for DUPLEX BAM File |

secondaryFiles:

| File ID | Example | Description |
| :--- | :--- | :--- |
| **duplex-bam** | seracare\_unfiltered-duplex.bai | index for collapsed unfiltered bam |
| **simplex-bam** | seracare\_unfiltered-simplex.bai | index for collapsed simplex bam |
| **unfiltered-bam** | seracare\_unfiltered\_abra\_fm.bai | index for collapsed bam |

Where:

* **&lt;seracare&gt;** refers to the Read 1 fastq basename \(w/o extension\)
  * Ex: SAMPLE\_R1\_001

